### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by IvanGrginovInf**
Changelog
---
* Fixed props test so it automatically tests all blocks based on the `props-output.js` file
* Added test for attribute overriding during registration (tested `getAttributes()`)
* Added `checkAttr` test so it always returns the attribute based on the available attributes / manifest defaults

### CODE REVIEW

File: tests/data/props-output.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /**
  * Define what props should or should not return on each lvl of a dependency tree, using the newName to identify 
  * dependencies.
+ *
+ * Ideally all blocks from the project should have a top lvl entry here with
```
Comment (goranalkovic-infinum) [inline tests/data/props-output.js:5]:
```suggestion
 * Ideally all blocks from the project should have a top level entry here with
```

File: tests/data/props-output.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+ * their dependency tree and expected / notExpected attributes + prefix defined.
```
Comment (goranalkovic-infinum) [inline tests/data/props-output.js:6]:
```suggestion
 * their dependency tree and `expected` / `notExpected` attributes and `prefix` defined.
```

File: tests/data/src/Blocks/components/mock-no-override/manifest.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+	"componentName": "mock-no-override",
+	"title": "Mock no override",
+	"description" : "Component used in the overriding attributes test. This one doesn't override any lvl but the block extending this component does.",
```
Comment (goranalkovic-infinum) [inline tests/data/src/Blocks/components/mock-no-override/manifest.json:4]:
```suggestion
	"description" : "Component used in the overriding attributes test. This one doesn't override any level but the block extending this component does.",
```

File: tests/unit/blocks/block-registration.test.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+			const blockExpectedProps = propsOutput[blockProps.realName];
+
+			recursiveAssertProps(blockExpectedProps, blockProps, true);
+		}
+	}
+});
+
+/**
+ * Unit tests for attribute overriding in parent blocks. Each block / component should be able
+ * to override it's child attributes.
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:199]:
```suggestion
 * to override its child attributes.
```

File: tests/unit/blocks/block-registration.test.js [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+	const lvl4Attributes = getAttributes(globalManifest, wrapperManifest, componentManifests, lvl4Manifest);
+	expect(lvl4Attributes).toHaveProperty('mockAttributeOverrideMockParagraphMockTypographyContent');
+	expect(lvl4Attributes.mockAttributeOverrideMockParagraphMockTypographyContent).toHaveProperty('default');
+	expect(lvl4Attributes.mockAttributeOverrideMockParagraphMockTypographyContent.default).toBe('This is lvl 4 override');
+});
+
+it('tests block / component attribute overriding when overriding happens on top and bottom lvl, but not in the middle component', () => {
+	const componentManifests = getAllComponentManifests();
+
+	// Test bottom lvl, mock paragraph extending mock typography
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:233]:
```suggestion
	// Test bottom level, mock paragraph extending mock typography
```

File: tests/unit/blocks/block-registration.test.js [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+	// Test lvl 3, component overriding attributes 1 lvl below it.
+	const lvl3Manifest = getMockComponentManifest('mock-attribute-override');
+	const lvl3Attributes = getAttributes(globalManifest, wrapperManifest, componentManifests, lvl3Manifest);
+
+	expect(lvl3Attributes).toHaveProperty('mockParagraphMockTypographyContent');
+	expect(lvl3Attributes.mockParagraphMockTypographyContent).toHaveProperty('default');
+	expect(lvl3Attributes.mockParagraphMockTypographyContent.default).toBe('This is lvl 3 override');
+
+	// Test lvl 4, component overriding attributes 1 lvl below it where attribute has already been overriden.
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:222]:
```suggestion
	// Test level 4: component overriding attributes 1 level below it where attribute has already been overridden.
```

File: tests/unit/blocks/block-registration.test.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+	// Test bottom lvl, mock paragraph extending mock typography
+	const lvl2Manifest = getMockComponentManifest('mock-paragraph');
+	const lvl2Attributes = getAttributes(globalManifest, wrapperManifest, componentManifests, lvl2Manifest);
+
+	expect(lvl2Attributes).toHaveProperty('mockTypographyContent');
+	expect(lvl2Attributes.mockTypographyContent).toHaveProperty('default');
+	expect(lvl2Attributes.mockTypographyContent.default).toBe('This is lvl 2 override');
+
+	// Test lvl 3, component overriding attributes 1 lvl below it.
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:214]:
```suggestion
	// Test level 3: component overriding attributes 1 level below it.
```

File: tests/unit/blocks/block-registration.test.js [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+/**
+ * Unit tests for attribute overriding in parent blocks. Each block / component should be able
+ * to override it's child attributes.
+ *
+ * @group unit
+ */
+it('tests block / component attribute generation and overriding attributes in parent component / blocks', () => {
+	const componentManifests = getAllComponentManifests();
+
+	// Test bottom lvl, mock paragraph extending mock typography
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:206]:
```suggestion
	// Test bottom level: mock paragraph extending mock typography
```

File: tests/unit/blocks/block-registration.test.js [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+	const lvl2Manifest = getMockComponentManifest('mock-paragraph');
+	const lvl2Attributes = getAttributes(globalManifest, wrapperManifest, componentManifests, lvl2Manifest);
+
+	expect(lvl2Attributes).toHaveProperty('mockTypographyContent');
+	expect(lvl2Attributes.mockTypographyContent).toHaveProperty('default');
+	expect(lvl2Attributes.mockTypographyContent.default).toBe('This is lvl 2 override');
+
+	// Test lvl 4, component overriding attributes 1 lvl below it where attribute has already been overriden.
```
Comment (goranalkovic-infinum) [inline tests/unit/blocks/block-registration.test.js:241]:
```suggestion
	// Test level 4: component overriding attributes 1 level below it where attribute has already been overridden.
```

**Review by goranalkovic-infinum — APPROVED** [review]
Left minor comments, great job!

**Review by dingo-d — APPROVED** [review]

**Pull request closed without merging**